test/basicfuns.jl

```diff
@@ -137,6 +137,10 @@ end
     @test isnan(logsumexp([NaN, 9.0]))
     @test isnan(logsumexp([NaN, Inf]))
     @test isnan(logsumexp([NaN, -Inf]))
+
+    # logsumexp with general iterables (issue #63)
+    xs = range(-500, stop = 10, length = 1000)
+    @test logsumexp(x for x in xs) == logsumexp(xs)
 end
```

Comment:
@stevengj mentioned [here](https://discourse.julialang.org/t/fast-logsumexp/22827/7) that underflow can also be a problem for `logsumexp` implementations, e.g. with `[1e-20, log(1e-20)]`. It seems that your PR handles this correctly (returning `1.99...e-20`, while the previous one incorrectly returned `1.0e-20`. Maybe add a test above around line 105 to cover this case?

Reply:
Sure, I added one.